### PR TITLE
Fixes #23651 - correct binary dependencies

### PR DESCRIPTION
**Pull request by mmalchuk**
Since 2.0.2 version ruby-smart-proxy-ansible package already contains
the foreman_ansible_core gem so it should set correct dependencies
with the ruby-foreman-ansible-core package.

Signed-off-by: Maksim Malchuk <maksim.malchuk@gmail.com>

For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [x] 1.17
* [ ] 1.16
* [ ] 1.15

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ekohl** [comment]
Thanks for the patch but I wonder if the actual bug is that core isn't properly excluded. @mmoll?

**Comment by mmoll** [comment]
Yes, the actual bug is in the gem that was uploaded to rubygems.org and it needs to get fixed there.

**Comment by mmalchuk** [comment]
@mmoll until then this fix will solve the issue with broken 1.17 smart-proxy-ansible on binary Ubuntu installations.

**Comment by mmoll** [comment]
@dLobatog @bastilian please release a new, clean build of the smart_proxy_ansible gem

**Comment by mmalchuk** [comment]
@mmoll what do you mean by saying "clean"? revert of the https://github.com/theforeman/smart_proxy_ansible/commit/1f52708d51e5ca909e21c54da20aadc1ff3381a2 ?

**Comment by mmoll** [comment]
uhm, I have no idea, what happended, there, @dLobatog could you please clarify?

File: plugins/smart_proxy_ansible/control [hunk 1]
```diff
@@ -11,7 +11,9 @@ Package: ruby-smart-proxy-ansible
 Architecture: all
 XB-Ruby-Versions: ${ruby:Versions}
 Depends: ${shlibs:Depends}, ${misc:Depends}, ruby | ruby-interpreter, foreman-proxy (>= 1.11.0~rc3),
-  ruby-foreman-ansible-core (>= 2.0.2), ruby-foreman-ansible-core (<< 3.0.0),
   ruby-smart-proxy-dynflow (>= 0.1.0), ruby-smart-proxy-dynflow (<< 0.2.0),
   ansible (>= 2.2.0)
+Provides: ruby-foreman-ansible-core (= 2.0.2)
```
[the remark this answers is not in the document]
Comment (mmalchuk) [reply plugins/smart_proxy_ansible/control]: new patch ready

**Comment by mmalchuk** [comment]
@dLobatog please have a look at @theforeman-bot message. you change the ticket to the wrong project.

**Comment by mmalchuk** [comment]
closed in favor of GH-2514

**Pull request closed without merging**